src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/resample_ref.cl

```diff
@@ -318,6 +320,88 @@ KERNEL (resample_gpu_ref)(__global INPUT0_TYPE* input,
+
+    bool FrontTopLOutOfBounds = in_z2 < 0 || in_z2 >= in_size[2] || in_y1 < 0 || in_y1 >= in_size[3] || in_x1 < 0 || in_x1 >= in_size[4];
+    bool FrontTopROutOfBounds = in_z2 < 0 || in_z2 >= in_size[2] || in_y1 < 0 || in_y1 >= in_size[3] || in_x2 < 0 || in_x2 >= in_size[4];
+    bool FrontBottomLOutOfBounds = in_z2 < 0 || in_z2 >= in_size[2] || in_y2 < 0 || in_y2 >= in_size[3] || in_x1 < 0 || in_x1 >= in_size[4];
+    bool FrontBottomROutOfBounds = in_z2 < 0 || in_z2 >= in_size[2] || in_y2 < 0 || in_y2 >= in_size[3] || in_x2 < 0 || in_x2 >= in_size[4];
+
+    OUTPUT_TYPE x111 = BackTopLOutOfBounds ? INPUT0_VAL_ZERO :input[INPUT0_GET_INDEX(batch, feature, in_z1, in_y1, in_x1)];
+    OUTPUT_TYPE x211 = BackTopROutOfBounds ? INPUT0_VAL_ZERO :input[INPUT0_GET_INDEX(batch, feature, in_z1, in_y1, in_x2)];
+    OUTPUT_TYPE x121 = BackBottomLOutOfBounds ? INPUT0_VAL_ZERO :input[INPUT0_GET_INDEX(batch, feature, in_z1, in_y2, in_x1)];
+    OUTPUT_TYPE x221 = BackBottomROutOfBounds ? INPUT0_VAL_ZERO :input[INPUT0_GET_INDEX(batch, feature, in_z1, in_y2, in_x2)];
+    OUTPUT_TYPE x112 = FrontTopLOutOfBounds ? INPUT0_VAL_ZERO :input[INPUT0_GET_INDEX(batch, feature, in_z2, in_y1, in_x1)];
+    OUTPUT_TYPE x212 = FrontTopROutOfBounds ? INPUT0_VAL_ZERO :input[INPUT0_GET_INDEX(batch, feature, in_z2, in_y1, in_x2)];
+    OUTPUT_TYPE x122 = FrontBottomLOutOfBounds ? INPUT0_VAL_ZERO :input[INPUT0_GET_INDEX(batch, feature, in_z2, in_y2, in_x1)];
+    OUTPUT_TYPE x222 = FrontBottomROutOfBounds ? INPUT0_VAL_ZERO :input[INPUT0_GET_INDEX(batch, feature, in_z2, in_y2, in_x2)];
+#else
+    OUTPUT_TYPE x111 = input[INPUT0_GET_INDEX(batch, feature, in_z1, in_y1, in_x1)];//BackTopLOutOfBounds
+    OUTPUT_TYPE x211 = input[INPUT0_GET_INDEX(batch, feature, in_z1, in_y1, in_x2)];//BackTopROutOfBounds
+    OUTPUT_TYPE x121 = input[INPUT0_GET_INDEX(batch, feature, in_z1, in_y2, in_x1)];//BackBottomLOutOfBounds
+    OUTPUT_TYPE x221 = input[INPUT0_GET_INDEX(batch, feature, in_z1, in_y2, in_x2)];//BackBottomROutOfBounds
+    OUTPUT_TYPE x112 = input[INPUT0_GET_INDEX(batch, feature, in_z2, in_y1, in_x1)];//FrontTopLOutOfBounds
+    OUTPUT_TYPE x212 = input[INPUT0_GET_INDEX(batch, feature, in_z2, in_y1, in_x2)];//FrontTopROutOfBounds
+    OUTPUT_TYPE x122 = input[INPUT0_GET_INDEX(batch, feature, in_z2, in_y2, in_x1)];//FrontBottomLOutOfBounds
+    OUTPUT_TYPE x222 = input[INPUT0_GET_INDEX(batch, feature, in_z2, in_y2, in_x2)];//FrontBottomROutOfBounds
+#endif
+
+    OUTPUT_TYPE interp_val = dx2 * dy2 * dz2 * x111 + dx1 * dy2 * dz2 * x211;
+    interp_val += dx2 * dy1 * dz2 * x121 + dx1 * dy1 * dz2 * x221;
+    interp_val += dx2 * dy2 * dz1 * x112 + dx1 * dy2 * dz1 * x212;
+    interp_val += dx2 * dy1 * dz1 * x122 + dx1 * dy1 * dz1 * x222;
+
+    int o_idx = OUTPUT_GET_INDEX(batch, feature, oz, oy, ox);
+    output[OUTPUT_GET_INDEX(batch, feature, oz, oy, ox)] = interp_val;
```

Comment:
Since this is a reference kernel and there is no any custom limitation for 5D linear onnx for fusions with other simple operations, we should provide support of fusing mechanism. Please take a look on HAS_FUSED_OPS macro and implement this feature in your kernel

Reply:
added as it done for 4d formats
is any test for that?

Reply:
See https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/tests/fusions/resample_fusion_test.cpp